formidling/src/main/java/no/nav/ung/sak/formidling/innhold/InnvilgelseInnholdBygger.java

```diff
@@ -44,219 +40,152 @@ public class InnvilgelseInnholdBygger implements VedtaksbrevInnholdBygger {
 
-    private static LocalDateSegment<GrunnlagOgTilkjentYtelse> sammenstillGrunnlagOgTilkjentYtelse(
-        LocalDateInterval di, LocalDateSegment<TilkjentYtelseVerdi> lhs, LocalDateSegment<UngdomsytelseSatser> rhs) {
-        var tilkjentYtelse = lhs.getValue();
-        var satsPerioder = rhs.getValue();
-        return new LocalDateSegment<>(di,
-            new GrunnlagOgTilkjentYtelse(
-                satsPerioder.satsType(),
-                tilHeltall(satsPerioder.grunnbeløp()),
-                tilFaktor(satsPerioder.grunnbeløpFaktor()),
-                tilHeltall(satsPerioder.grunnbeløp().multiply(satsPerioder.grunnbeløpFaktor())),
-                tilHeltall(satsPerioder.dagsats()),
-                satsPerioder.antallBarn(),
-                satsPerioder.dagsatsBarnetillegg(),
-                tilHeltall(tilkjentYtelse.dagsats()),
-                tilkjentYtelse.utbetalingsgrad()
-                ));
+        if (overgangLavSats) {
+            brevfeilHåndterer.registrerFeilmelding("Kan ikke ha overgang fra høy til lav sats men fant det mellom %s og %s".formatted(previous.getLocalDateInterval(), current.getLocalDateInterval()));
+        }
 
+        var totaltBarnetillegg = BigDecimal.valueOf(currentSatser.dagsatsBarnetillegg()).multiply(BigDecimal.valueOf(currentSatser.antallBarn()));
+
+        return new SatsEndringHendelseDto(
+            overgangTilHøySats,
+            fødselBarn,
+            dødsfallBarn,
+            current.getFom(),
+            tilHeltall(currentSatser.dagsats().add(totaltBarnetillegg)),
+            dødsfallBarn ? previousSatser.dagsatsBarnetillegg() : currentSatser.dagsatsBarnetillegg(),
+            fikkFlereBarn
+        );
     }
 
-    private static Set<GbeløpPeriodeDto> lagGbeløpPerioderDto(LocalDateTimeline<GrunnlagOgTilkjentYtelse> grunnlagOgTilkjentYtelseTimeline) {
-        return grunnlagOgTilkjentYtelseTimeline
-            .mapSegment(GrunnlagOgTilkjentYtelse::grunnbeløp)
-            .compress().stream()
-            .map(it -> new GbeløpPeriodeDto(
-                new PeriodeDto(it.getFom(), it.getTom()), it.getValue()))
+    private static SatsOgBeregningDto mapSatsOgBeregning(NavigableSet<LocalDateSegment<UngdomsytelseSatser>> satsSegments, BrevfeilHåndterer brevfeilHåndterer) {
+        var satser = satsSegments.stream()
+            .map(it -> it.getValue().satsType())
             .collect(Collectors.toSet());
-    }
 
-    private static SatserDto lagSatsDto(LocalDateTimeline<GrunnlagOgTilkjentYtelse> grunnlagOgTilkjentYtelseTimeline) {
-        List<GrunnlagOgTilkjentYtelse> sortertGrunnlagOgTilkjentYtelseVerdier = grunnlagOgTilkjentYtelseTimeline.stream()
-            .sorted(Comparator.comparing(LocalDateSegment::getLocalDateInterval, Comparator.reverseOrder())) // nyeste først
-            .map(LocalDateSegment::getValue)
-            .distinct()
-            .toList();
+        var kunHøySats = satser.size() == 1 && satser.contains(UngdomsytelseSatsType.HØY);
```

Comment:
Dette kan forenkles slik at du slipper å sjekke size og contains eksplisitt.
`Set.of(UngdomsytelseSatsType.HØY).equals(satser);`